File: src/main/java/cabalgatas_salento/reservas/mapper/ReservacionMapper.java
package cabalgatas_salento.reservas.mapper;

import cabalgatas_salento.reservas.dto.response.ReservacionResponse;
import cabalgatas_salento.reservas.entity.Reservacion;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class ReservacionMapper {

    private final SalidaMapper salidaMapper;
    private final ParticipanteMapper participanteMapper;

    public ReservacionResponse toResponse(Reservacion r) {
        return ReservacionResponse.builder()
                .id(r.getId())
                .salida(salidaMapper.toResponse(r.getSalida()))
                .clienteId(r.getCliente() != null ? r.getCliente().getId() : null)
                .clienteNombre(r.getCliente() != null
                        ? r.getCliente().getPrimerNombre() + " " + r.getCliente().getPrimerApellido()
                        : null)
                .adminId(r.getAdmin() != null ? r.getAdmin().getId() : null)
                .numPeople(r.getNumPeople())
                .precioUnitario(r.getPrecioUnitario())
                .total(r.getTotal())
                .estado(r.getEstado())
                .participantes(r.getParticipantes().stream()
                        .map(participanteMapper::toResponse)
                        .toList())
                .build();
    }
}

File: src/reservaciones/mappers/reservacion.mapper.ts
import { SalidaMapper } from '../../salidas/mappers/salida.mapper';
import { ReservacionResponseDto } from '../dto/reservacion-response.dto';
import { Reservacion } from '../entities/reservacion.entity';
import { ParticipanteMapper } from './participante.mapper';

export class ReservacionMapper {
  static toResponseDto(entity: Reservacion): ReservacionResponseDto {
    const dto = new ReservacionResponseDto();
    dto.id = entity.id;
    dto.salida = entity.salida ? SalidaMapper.toResponseDto(entity.salida) : null;
    dto.clienteId = entity.cliente?.id ?? null;
    dto.clienteNombre = entity.cliente
      ? `${entity.cliente.primerNombre} ${entity.cliente.primerApellido}`
      : null;
    dto.adminId = entity.admin?.id ?? null;
    dto.numPeople = entity.numPeople;
    dto.precioUnitario = entity.precioUnitario;
    dto.total = entity.total;
    dto.estado = entity.estado;
    dto.participantes = (entity.participantes ?? []).map(
      ParticipanteMapper.toResponseDto,
    );
    return dto;
  }
}
